ITKImageProcessingFilters/itkImportImageMallocContainer.hxx

```diff
@@ -0,0 +1,233 @@
+      else
+      {
+        m_Size = size;
+        this->Modified();
+      }
+    }
+    else
+    {
+      m_ImportPointer = this->AllocateElements( size, UseDefaultConstructor );
+      m_Capacity = size;
+      m_Size = size;
+      m_ContainerManageMemory = true;
+      this->Modified();
+    }
+  }
+
+  /**
+  * Tell the container to try to minimize its memory usage for storage of
+  * the current number of elements.
+  */
+  template< typename TElementIdentifier, typename TElement >
+  void
+    ImportImageMallocContainer< TElementIdentifier, TElement >
+    ::Squeeze( void )
+  {
+    if( m_ImportPointer )
+    {
+      if( m_Size < m_Capacity )
+      {
+        const TElementIdentifier size = m_Size;
+        TElement *               temp = this->AllocateElements( size, false );
+        std::copy( m_ImportPointer,
+          m_ImportPointer + m_Size,
+          temp );
+
+        DeallocateManagedMemory();
+
+        m_ImportPointer = temp;
+        m_ContainerManageMemory = true;
+        m_Capacity = size;
+        m_Size = size;
+
+        this->Modified();
+      }
+    }
+  }
+
+  /**
+  * Tell the container to try to minimize its memory usage for storage of
+  * the current number of elements.
+  */
+  template< typename TElementIdentifier, typename TElement >
+  void
+    ImportImageMallocContainer< TElementIdentifier, TElement >
+    ::Initialize( void )
+  {
+    if( m_ImportPointer )
+    {
+      DeallocateManagedMemory();
+
+      m_ContainerManageMemory = true;
+
+      this->Modified();
+    }
+  }
+
+  /**
+  * Set the pointer from which the image data is imported.  "num" is
+  * the number of pixels in the block of memory. If
+  * "LetContainerManageMemory" is false, then the application retains
+  * the responsibility of freeing the memory for this image data.  If
+  * "LetContainerManageMemory" is true, then this class will free the
+  * memory when this object is destroyed.
+  */
+  template< typename TElementIdentifier, typename TElement >
+  void
+    ImportImageMallocContainer< TElementIdentifier, TElement >
+    ::SetImportPointer( TElement *ptr, TElementIdentifier num,
+    bool LetContainerManageMemory )
+  {
+    DeallocateManagedMemory();
+    m_ImportPointer = ptr;
+    m_ContainerManageMemory = LetContainerManageMemory;
+    m_Capacity = num;
+    m_Size = num;
+
+    this->Modified();
+  }
+
+  template< typename TElementIdentifier, typename TElement >
+  TElement *ImportImageMallocContainer< TElementIdentifier, TElement >
+    ::AllocateElements( ElementIdentifier size, bool UseDefaultConstructor ) const
+  {
+  // Encapsulate all image memory allocation here to throw an
+  // exception when memory allocation fails even when the compiler
+  // does not do this by default.
+  TElement *data;
+
+  try
+    {
+    data = (TElement*)(malloc( sizeof( TElement ) * size ));
+    if ( UseDefaultConstructor )
+      {
+        new (data)TElement();//POD types initialized to 0, others use default constructor.
```

Comment:
This statement is there to call the default constructor ([see here](http://stackoverflow.com/questions/2995099/malloc-and-constructors))